GetandTake/Areas/Identity/Pages/Account/Manage/Email.cshtml.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Text;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Identity.UI.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.WebUtilities;
+
+namespace GetandTake.Areas.Identity.Pages.Account.Manage;
+
+public class EmailModel : PageModel
+{
+    private readonly UserManager<IdentityUser> _userManager;
+    private readonly SignInManager<IdentityUser> _signInManager;
+    private readonly IEmailSender _emailSender;
+
+    public EmailModel(
+        UserManager<IdentityUser> userManager,
+        SignInManager<IdentityUser> signInManager,
+        IEmailSender emailSender)
+    {
+        _userManager = userManager;
+        _signInManager = signInManager;
+        _emailSender = emailSender;
+    }
+
+    public string Email { get; set; }
+
+    public bool IsEmailConfirmed { get; set; }
+
+    [TempData]
+    public string StatusMessage { get; set; }
+
+    [BindProperty]
+    public InputModel Input { get; set; }
+
+    public class InputModel
+    {
+        [Required]
+        [EmailAddress]
+        [Display(Name = "New email")]
+        public string NewEmail { get; set; }
+    }
+
+    private async Task LoadAsync(IdentityUser user)
+    {
+        var email = await _userManager.GetEmailAsync(user);
+        Email = email;
+
+        Input = new InputModel
+        {
+            NewEmail = email,
+        };
+
+        IsEmailConfirmed = await _userManager.IsEmailConfirmedAsync(user);
+    }
+
+    public async Task<IActionResult> OnGetAsync()
+    {
+        var user = await _userManager.GetUserAsync(User);
+        
+        if (user == null)
+        {
+            return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+        }
+
+        await LoadAsync(user);
+        
+        return Page();
+    }
+
+    public async Task<IActionResult> OnPostChangeEmailAsync()
+    {
+        var user = await _userManager.GetUserAsync(User);
+        
+        if (user == null)
+        {
+            return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+        }
+
+        if (!ModelState.IsValid)
+        {
+            await LoadAsync(user);
+            
+            return Page();
+        }
+
+        var email = await _userManager.GetEmailAsync(user);
+        
+        if (Input.NewEmail != email)
```

Comment:
Use `string.Equals` for strings comparison, please

Reply:
!string.Equals(Input.NewEmail,email)